### PR TITLE
feat: add microsoft teams integration

### PR DESCRIPTION
**Pull request by sascha-andres**
Add ability to configure Microsoft Teams integration

### CODE REVIEW

File: services.go [hunk 1]
```diff
@@ -513,3 +513,90 @@ func (s *ServicesService) DeleteJenkinsCIService(pid interface{}, options ...Opt
...
+	Service
+	Properties *MicrosoftTeamsServiceProperties `json:"properties"`
+}
+
+// MicrosoftTeamsServiceProperties represents Microsoft Teams specific properties.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/services.html#microsoft-teams
+type MicrosoftTeamsServiceProperties struct {
+	WebHook                   *string `webhook:"webhook,omitempty" json:"webhook,omitempty"`
```
Comment (svanharmelen) [inline services.go]: It looks like you didn’t `gofmt` your code. Could you please do that to make sure its formatted correctly?

File: services.go [hunk 2]
```diff
@@ -513,3 +513,90 @@ func (s *ServicesService) DeleteJenkinsCIService(pid interface{}, options ...Opt
...
+	}
+	u := fmt.Sprintf("projects/%s/services/microsoft-teams", url.QueryEscape(project))
+
+	req, err := s.client.NewRequest("DELETE", u, nil, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(req, nil)
+}
```
Comment (svanharmelen) [inline services.go]: Please end the file with a newline... Maybe `gofmt` does this as well, but not sure about that.

File: services.go (hunk 1 above)
Comment (svanharmelen) [reply services.go]: You should not use a pointer here and the tag is not correct either... Please make sure they are inline with the other examples in this file.

File: services.go [hunk 3]
```diff
@@ -513,3 +513,90 @@ func (s *ServicesService) DeleteJenkinsCIService(pid interface{}, options ...Opt
...
+
+	return svc, resp, err
+}
+
+// SetMicrosoftTeamsServiceOptions represents the available SetMicrosoftTeamsService()
+// options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/services.html#create-edit-microsoft-teams-service
+type SetMicrosoftTeamsServiceOptions MicrosoftTeamsServiceProperties
```
Comment (svanharmelen) [inline services.go]: This should be a different type with a pointer to the `webhook` value and different tags.

**Comment by sascha-andres** [comment]
@svanharmelen changes done

File: services.go [hunk 4]
```diff
@@ -513,3 +513,91 @@ func (s *ServicesService) DeleteJenkinsCIService(pid interface{}, options ...Opt
...
+	Service
+	Properties *MicrosoftTeamsServiceProperties `json:"properties"`
+}
+
+// MicrosoftTeamsServiceProperties represents Microsoft Teams specific properties.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/services.html#microsoft-teams
+type MicrosoftTeamsServiceProperties struct {
+	WebHook string `url:"webhook,omitempty" json:"webhook,omitempty"`
```
[the remark this answers is not in the document]
Comment (sascha-andres) [reply services.go]: changed

**Pull request merged**